resident/resident-service/src/main/java/io/mosip/resident/service/impl/ProxyMasterdataServiceImpl.java

```diff
@@ -413,5 +462,118 @@ public ResponseWrapper<?> getGenderTypesByLangCode(String langCode) throws Resid
+		InputStream downLoadRegCenterTemplateData = templateManager.merge(downLoadRegCenterTemplate, regCentersMap);
+		
+		StringWriter writer = new StringWriter();
+		IOUtils.copy(downLoadRegCenterTemplateData, writer, "UTF-8");
+		ByteArrayOutputStream pdfValue = (ByteArrayOutputStream) pdfGenerator.generate(writer.toString());
+		logger.debug("ResidentServiceImpl::residentServiceHistoryPDF()::exit");
+		return convertOutputStreamToInputStream(pdfValue);
+	}
+	
+	/**
+	 * update the registrationcenter details
+	 */
+	private void addRegistrationCenterDtls(int index, RegistrationCenterDto regCenterDto)
+			throws ResidentServiceCheckedException, Exception {
+		String workingHours = "";
+		String fullAddress = getFullAddress(regCenterDto.getAddressLine1(), regCenterDto.getAddressLine2(),
+				regCenterDto.getAddressLine3());
+		regCenterDto.setSerialNumber(index+1);
+		regCenterDto.setFullAddress(fullAddress);
+		List<WorkingDaysDto>  workingDaysList = getRegCenterWorkingDays(regCenterDto.getId(), regCenterDto.getLangCode());
+		workingHours = workingDaysList.get(0).getName() + "-" + workingDaysList.get(1).getName() + "|"
+				+ regCenterDto.getCenterStartTime().substring(0, 5) + " " + "am" + "-"
+				+ regCenterDto.getCenterEndTime().substring(0, 5) + " " + "pm";
+		regCenterDto.setWorkingHours(workingHours);
+	}
+	/**
+	 * convert output stream to input stream
+	 * @param orgByteOutStream
+	 * @return
+	 */
+	private static InputStream convertOutputStreamToInputStream(ByteArrayOutputStream orgByteOutStream) {
+		PipedInputStream in = new PipedInputStream();
+		new Thread(new Runnable() {
+			public void run() {
+				try (final PipedOutputStream out = new PipedOutputStream(in)) {
+					orgByteOutStream.writeTo(out);
+				} catch (IOException e) {
+					logger.debug("convert Output stream to input stream" +e.getMessage());
+				}
+			}
+		}).start();
+		return in;
+	}
+	/**
+	 * return the full address
+	 * @param address1
+	 * @param address2
+	 * @param address3
+	 * @return
+	 */
+	private String getFullAddress(String address1, String address2, String address3) {
+		StringBuilder fullAddress = new StringBuilder();
+		fullAddress.append(address1 + "," + address2 + "," + address3);
+		return fullAddress.toString();
+	}
+	
+	/**
+	 * return the starting and ending working day details
+	 * @param regCenterId
+	 * @param langCode
+	 * @return
+	 * @throws ResidentServiceCheckedException
+	 * @throws Exception
+	 */
+	private List<WorkingDaysDto> getRegCenterWorkingDays(String regCenterId, String langCode) throws ResidentServiceCheckedException,Exception {
+		ResponseWrapper<?> responseWrapper;
+		responseWrapper = proxyMasterdataService.getRegistrationCenterWorkingDays(regCenterId, langCode);
+		WorkingDaysResponseDto workingDaysResponeDtls = mapper.readValue(mapper.writeValueAsString(responseWrapper.getResponse()), WorkingDaysResponseDto.class);		
+		List<WorkingDaysDto> workingDaysList=workingDaysResponeDtls.getWorkingdays();
```

Comment:
How big is this data? Will it not be too big?

Reply:
very less data. Only get 6 working days of Data